tests/mock/test_functions.py

```diff
@@ -289,3 +294,22 @@ def test_to_char_is_row_index_agnostic(session):
     assert df.filter(col("a") > 3).select(to_char(col("a")), col("b")).collect() == [
         Row("5", 6)
     ]
+
+
+@pytest.mark.skipif(
+    "not config.getoption('local_testing_mode', default=True)",
+    reason="Only test local testing code in local testing mode.",
+)
+def test_function_mock_and_call_neg(session):
```

Comment:
Hi, I do not have a tons of knowledge here, but as far as I see it, there is no tests for register() and unregister() function right? Can we add some unit tests for new functions in this PR?